packages/flutter_tools/lib/src/web/devfs_config.dart

```diff
@@ -0,0 +1,261 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+// import 'package:source_span/source_span.dart';
+import 'package:yaml/yaml.dart';
+
+import '../base/common.dart';
+import '../base/file_system.dart' as fs;
+import '../globals.dart' as globals;
+import 'devfs_proxy.dart';
+
+const devConfigFilePath = 'web_dev_config.yaml';
+
+@immutable
+class WebDevServerConfig {
+  const WebDevServerConfig({
+    this.headers = const <String, String>{},
+    this.host = 'any',
+    this.port,
+    this.https,
+    this.proxy = const <ProxyRule>[],
+  });
+
+  factory WebDevServerConfig.fromYaml(YamlMap yaml) {
+    final headers = <String, String>{};
+    if (yaml['headers'] != null) {
+      if (yaml['headers'] is! YamlList) {
+        throwToolExit('Headers must be a List of maps. Found ${yaml['headers'].runtimeType}');
+      }
+      final headersList = yaml['headers'] as YamlList;
+      for (final dynamic item in headersList) {
+        if (item is! YamlMap) {
+          throwToolExit(
+            'Each header entry must be a map with "name" and "value" keys. Found ${item.runtimeType}',
+          );
+        }
+        final YamlMap headerMap = item;
+        if (!headerMap.containsKey('name') || !headerMap.containsKey('value')) {
+          throwToolExit('Each header entry must contain "name" and "value" keys.');
+        }
+        final dynamic name = headerMap['name'];
+        final dynamic value = headerMap['value'];
+
+        if (name is! String || value is! String) {
+          throwToolExit(
+            'Header "name" and "value" must be strings. Found name: ${name.runtimeType}, value: ${value.runtimeType}',
+          );
+        }
+        headers[name] = value;
+      }
+    }
+    if (yaml['host'] is! String && yaml['host'] != null) {
+      throwToolExit('Host must be a String. Found ${yaml['host'].runtimeType}');
```

Comment:
The key is `host` not `Host`, so we should be consistent and use the lowercase string in the error message (here and elsewhere in this file).